### PR TITLE
Make a review

### PR DESCRIPTION
**Pull request by limkimhuor**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  def show
+  end
+
+  def new
+  end
+
+  def create
+    @review = current_user.reviews.build review_params
+    if @review.save
+      flash[:success] = t("flashs.success")
```
Comment (RathanakSreang) [inline app/controllers/reviews_controller.rb]: remove ()

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      redirect_to request.referer || root_url
+    else
+      flash[:danger] = t("flashs.notblank")
```
Comment (RathanakSreang) [inline app/controllers/reviews_controller.rb]: remove ()

File: app/views/admins/sessions/new.html.erb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <div class="form-area col-sm-4 col-sm-offset-4">
   <h2><%= t "headers.signin" %></h2>
-
   <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+  <%= render 'shared/error_messages', object: f.object %>
```
Comment (RathanakSreang) [inline app/views/admins/sessions/new.html.erb]: use `"` instead of `'`

File: app/views/reviews/_review_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<%= form_for(@review) do |f| %>
+<%= render 'shared/error_messages', object: f.object %>
+<div><%= f.hidden_field :book_id, value: book.id %></div>
+<div class="field">
+  <%= f.text_area :review, placeholder: "Compose new review", class: 'form-control' %>
+</div>
+  <%= f.submit t("buttons.review"), class: "btn btn-primary" %>
+<% end %>
```
Comment (RathanakSreang) [inline app/views/reviews/_review_form.html.erb]: new line

File: app/views/reviews/_review_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<%= form_for(@review) do |f| %>
+<%= render 'shared/error_messages', object: f.object %>
```
Comment (RathanakSreang) [inline app/views/reviews/_review_form.html.erb]: use `"` instead of `'`

File: app/views/reviews/_review_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<%= form_for(@review) do |f| %>
+<%= render "shared/error_messages", object: f.object %>
+<div><%= f.hidden_field :book_id, value: book.id %></div>
+<div class="field">
+  <%= f.text_area :review, placeholder: "Compose new review", class: 'form-control' %>
```
Comment (RathanakSreang) [inline app/views/reviews/_review_form.html.erb]: use`"`

**Comment by RathanakSreang** [comment]
Ok

**Comment by HorNorin** [comment]
OK

File: app/controllers/reviews_controller.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  before_action :authenticate_user!, only: [:create, :destroy]
+
+  def create
+    @review = current_user.reviews.build review_params
+    if @review.save
+      flash[:success] = t "flashs.success"
+      redirect_to request.referer || root_url
+    else
+      flash[:danger] = t "flashs.notblank"
+      redirect_to request.referer || root_url
```
Comment (chuanhtuan) [inline app/controllers/reviews_controller.rb]: duplicate code

File: app/views/reviews/_review_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<%= form_for(@review) do |f| %>
```
Comment (chuanhtuan) [inline app/views/reviews/_review_form.html.erb]: ()

**Pull request closed without merging**